book/src/languages.md

```diff
@@ -91,31 +95,97 @@ with the following priorities:
    replaced at runtime with the appropriate path separator for the operating
    system, so this rule would match against `.git\config` files on Windows.
 
-### Language Server configuration
+## Language Server configuration
+
+Language servers are configured separately in the table `language-server` in the same file as the languages `languages.toml`
+
+For example:
+
+```toml
+[language-server.mylang-lsp]
+command = "mylang-lsp"
+args = ["--stdio"]
+config = { provideFormatter = true }
+environment = { "ENV1" = "value1", "ENV2" = "value2" }
+
+[language-server.efm-lsp-prettier]
+command = "efm-langserver"
+
+[language-server.efm-lsp-prettier.config]
+documentFormatting = true
+languages = { typescript = [ { formatCommand ="prettier --stdin-filepath ${INPUT}", formatStdin = true } ] }
+```
```

Comment:
I am confused by the duplication of the `config` option. In the provided example:
```
[language-server.mylang-lsp]
command = "mylang-lsp"
args = ["--stdio"]
config = { provideFormatter = true }   <<<<<<<<<<<<<<<<<<<<<<<<< old format?
environment = { "ENV1" = "value1", "ENV2" = "value2" }

[language-server.efm-lsp-prettier.config] <<<<<<<<<<<<<<<<<<<<<<<<< new format?
documentFormatting = true
languages = { typescript = [ { formatCommand ="prettier --stdin-filepath ${INPUT}", formatStdin = true } ] }
```
Why is `config =` not considered legacy and removed? I would assume the new way to add a language server config would be:
```
[language-server.mylang-lsp]
command = "mylang-lsp"
args = ["--stdio"]
environment = { "ENV1" = "value1", "ENV2" = "value2" }

[language-server.mylang-lsp.config]
provideFormatter = true
```

Reply:
This is just TOML. They are equivalent ways to express a subtable.

Reply:
OK, then I don't think we should show both as it is just confusing, I would go with:
```
[language-server.mylang-lsp]
command = "mylang-lsp"
args = ["--stdio"]
environment = { "ENV1" = "value1", "ENV2" = "value2" }

[language-server.mylang-lsp.config]
provideFormatter = true
```

Reply:
Yeah I'm not completely sure about the formatting of language server configuration yet TBH. See also https://github.com/helix-editor/helix/pull/2507#discussion_r1140325081

Reply:
I don't think it's confusing, personally. At least not more confusing than TOML is in general. It's just a style choice really. You use inline if you want more compact, and if it gets too big, you turn it into a section.